### PR TITLE
Fix CActiveMasternode

### PR DESCRIPTION
**Pull request by UdjinM6**
- should run `ManageStateLocal` only when `eType == MASTERNODE_LOCAL`
- should set `nState = ACTIVE_MASTERNODE_NOT_CAPABLE` in `ManageStateInitial` to let `GetStatus` return proper message
- more/better log output
- _side work - added `GetStateString` helpers_

EDIT:
- slightly refactored `SendMasternodePing`

### CODE REVIEW

File: src/activemasternode.cpp [hunk 1]
```diff
@@ -190,48 +187,56 @@ void CActiveMasternode::ManageStateInitial()
...
-             GetType(),
-             fPingerEnabled,
-             pubKeyMasternode.GetID().ToString());
+             GetStatus(), fPingerEnabled, GetTypeString(), pubKeyMasternode.GetID().ToString());
 
     mnodeman.CheckMasternode(pubKeyMasternode);
-    masternode_info_t infoMn = mnodeman.GetMasternodeInfo(pubKeyMasternode);
-    if(infoMn.fInfoValid) {
-        if(infoMn.nProtocolVersion != PROTOCOL_VERSION) {
+    CMasternode mn;
```
Comment (tgflynn) [inline src/activemasternode.cpp]: Why did you change this ?  GetMasternodeInfo is more efficient because it avoids copying lots of data we don't usually need.

File: src/activemasternode.cpp [hunk 2]
```diff
@@ -95,28 +90,29 @@ std::string CActiveMasternode::GetType()
     return strType;
 }
 
-bool CActiveMasternode::SendMasternodePing(std::string& strErrorRet)
+bool CActiveMasternode::SendMasternodePing()
 {
-    if(vin == CTxIn()) {
+    if(fPingerEnabled) {
```
Comment (tgflynn) [inline src/activemasternode.cpp]: Looks like the logic is reversed here.

File: src/activemasternode.cpp (hunk 1 above)
Comment (UdjinM6) [inline src/activemasternode.cpp]: I needed `mn.GetStateString()` for a more precise error message, shouldn't add a lot of overhead here imo, `ManageStateRemote` is called initially and then only every `MASTERNODE_MIN_MNP_SECONDS`.

Comment (tgflynn) [inline src/activemasternode.cpp]: Maybe you should use a static function for the state to string conversion.

GetMasternodeInfo() is important.  We need to move away from non thread-safe access to the mnodeman data structures and I think this is the easiest way to do it for now.  I don't want to have to copy a whole MN object every time we need to lookup basic MN information and the only other way is to move to dynamic allocation and shared pointers, which would require much bigger changes.

Comment (UdjinM6) [inline src/activemasternode.cpp]: hmmm.. makes sense, will revert and implement another static one
